### DISCUSSION TITLE
Getting current job id

### DISCUSSION THREAD

**Discussion by gleseur** [sidekiq/sidekiq#5082] [Q&A] [answered]
Hi,

I'd like to be able to get the current job id from anywhere in the code for logging and tracing purpose.
Our use case is that we create audit records when some record are updated and it'd be super useful to be able to trace it to a particular jid to match that with the logs.

Currently we do that with the request id when the change directly comes from a controller, but would like to also have the sidekiq job id when the change comes from a job

Is there any officially accessible info on the current thread for this ?

Thanks,

**Comment by mperham** [comment, marked as answer]
While running inside a Sidekiq job, you can run this to get the current JID:

```ruby
Sidekiq::Context.current["jid"]
```

**Reply by gleseur** [reply]
Thank you @mperham That's it.
